public/js/graph/react_graph.js

```diff
@@ -942,4 +1008,138 @@ var Edge = React.createClass({
     }
 });
 
+
+var InfoBox = React.createClass({
+		getInitialState: function () {
+        return {
+            xPos: '0',
+            yPos: '0',
+            nodeId: '',
+            showInfobox: false
+        };
+    },
+
+    render: function () {
+        //TODO: move to CSS
+        var gStyles = {
+            cursor: 'pointer',
+            transition: 'opacity .4s',
+            opacity: this.state.showInfobox ? 1 : 0
+        }
+        var rectAttrs = {
+            id:this.state.nodeId+'-tooltip' + '-rect',
+            x: this.state.xPos,
+            y: this.state.yPos,
+            rx: '4',
+            ry: '4',
+            fill: 'white',
+            stroke: 'black',
+            'stroke-width': '2',
+            width: '60',
+            height: '30'
+        };
+
+        var textAttrs = {
+        	'id': this.state.nodeId +'-tooltip' + '-text',
+        	'x': parseFloat(this.state.xPos) + 60 / 2 - 18,
+        	'y': parseFloat(this.state.yPos) + 30 / 2 + 6
+        };
+
+        return (
+            this.state.showInfobox
+            ? <g id='infobox' className='tooltip-group' style={gStyles} {... this.props}>
+                <rect {... rectAttrs} ></rect>
+                <text {... textAttrs} >
+                    Info
+                </text>
+            </g>
```

Comment:
Generally don't use the ternary operator for multiple lines. Instead, use an `if`. Also, this check should happen at the very beginning of the function.